lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -4721,12 +4721,22 @@ ConstraintResult GenericSignatureBuilder::addTypeRequirement(
         // type that makes sense to use here, but, in practice, all
         // getLookupConformanceFns used in here don't use that parameter anyway.
         auto dependentType = CanType();
-        auto conformance =
-            getLookupConformanceFn()(dependentType, subjectType, proto->getDecl());
 
-        // FIXME: diagnose if there's no conformance.
-        if (conformance) {
-          addConditionalRequirements(*this, *conformance, inferForModule);
+        auto D = getGenericParams().front()->getDecl();
```

Comment:
Yes, I am proceeding with the conformance lookup if the subject type has a null decl context or isn't a nominal type declaration context, or otherwise isn't the type we're extending or declaring. Simply put, I am just making sure not to look up conformances if the base type of the subject and the current context match.